From: Filipe David Borba Manana <fdmanana@gmail.com>
Date: Tue, 20 Aug 2013 17:51:48 +0100
Git-commit: bdab49d760c14dd7d5a8f64dbfedd7a3f1121e9b
Patch-mainline: v3.12-rc2
References: FATE#312888
Subject: [PATCH] Btrfs: fix printing of non NULL terminated string

The name buffer is not terminated by a '\0' character,
therefore it needs to be printed with %.*s and use the
length of the buffer.

Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/delayed-inode.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/btrfs/delayed-inode.c
+++ b/fs/btrfs/delayed-inode.c
@@ -1485,10 +1485,10 @@ int btrfs_insert_delayed_dir_index(struc
 	mutex_lock(&delayed_node->mutex);
 	ret = __btrfs_add_delayed_insertion_item(delayed_node, delayed_item);
 	if (unlikely(ret)) {
-		printk(KERN_ERR "err add delayed dir index item(name: %s) into "
+		printk(KERN_ERR "err add delayed dir index item (name: %.*s) into "
 				"the insertion tree of the delayed node"
 				"(root id: %llu, inode id: %llu, errno: %d)\n",
-				name,
+				name_len, name,
 				(unsigned long long)delayed_node->root->objectid,
 				(unsigned long long)delayed_node->inode_id,
 				ret);
